### PR TITLE
LLVM emitter stage 1 step 2

### LINKED ISSUE #10: Implement call insn in nback.llvm
There's a stub that needs to be implemented.

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Implement  LLVM emitter for stage 1 step 2 and corresponding test cases
resolve #10 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/nback.llvm/llvm.bal [hunk 1]
```diff
@@ -257,8 +272,27 @@ public class Builder {
...
     public function call(Function fn, Value[] args) returns Value? {
-        // XXX todo
-        return ();
+        if fn.paramCount() != args.length() {
+            panic error(string `Number of arguments is invalid for function ${fn.ref()}`);
+        }
+        BasicBlock bb = self.bb();
+        string[] argStringContent = [];
+        foreach Value arg in args {
+            argStringContent.push(" ".'join(typeToString(arg.ty), arg.operand));
```
Comment (jclark) [inline modules/nback.llvm/llvm.bal]: Please don't use join here. Instead do it as I did in `header`.

File: modules/nback.llvm/llvm.bal [hunk 2]
```diff
@@ -257,8 +272,27 @@ public class Builder {
...
+        }
+        string argString = ", ".'join(...argStringContent);
+        if fn.returnsValue() {
+            string reg = bb.func.genReg();
+            bb.addInsn(reg, "=", "call", typeToString(fn.getReturnType()), fn.ref(), "(", argString, ")");
+            RetType returnType = fn.getReturnType();
+            if returnType is Type{
```
Comment (jclark) [inline modules/nback.llvm/llvm.bal]: Missing space before {

**Review by jclark — CHANGES_REQUESTED** [review]

File: modules/nback.llvm/llvm.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply modules/nback.llvm/llvm.bal]: Fixed

File: modules/nback.llvm/llvm.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply modules/nback.llvm/llvm.bal]: Fixed

File: compiler/modules/nback.llvm/llvm.bal [hunk 3]
```diff
@@ -257,8 +272,47 @@ public class Builder {
...
+            }
+            argStringContent.push(typeToString(arg.ty));
+            argStringContent.push(arg.operand);
+        }
+        if fn.returnsValue() {
+            string reg = bb.func.genReg();
+            string[] instruction = [];
+            instruction.push(reg);
+            instruction.push("=");
+            instruction.push("call");
```
Comment (jclark) [inline compiler/modules/nback.llvm/llvm.bal]: This code can be improved.  You have 6 lines repeated in both branches of the if. Also it is not necessary to have the temporary argString content array. `instruction` is not a good name: should be something plural like insnWords or words.

Comment (heshanpadmasiri) [reply compiler/modules/nback.llvm/llvm.bal]: Fixed

**Pull request merged**